### PR TITLE
Added collapse specification property

### PR DESCRIPTION
**Pull request by svenbru**
Added the ability to configure the collapse specification property in the SharePoint data source, like mentioned in older issues like #2135 and #911 

CollapseSpecification

### CODE REVIEW

**Comment by wobba** [comment]
Care to add it for the Graph provider as well? It's on the beta endpoint named `collapseProperties`.

https://learn.microsoft.com/en-us/graph/api/resources/searchrequest?view=graph-rest-beta

**Comment by svenbru** [comment]
@wobba Sure! I made the changes for the Graph provider.

File: search-parts/src/dataSources/MicrosoftSearchDataSource.ts [hunk 1]
```diff
@@ -787,6 +839,10 @@ export class MicrosoftSearchDataSource extends BaseDataSource<IMicrosoftSearchDa
             searchRequest.trimDuplicates = this.properties.trimDuplicates;
         }
 
+        if (this.properties.collapseProperties.length > 0) {
```
Comment (wobba) [inline search-parts/src/dataSources/MicrosoftSearchDataSource.ts:842]: This should go under the `this.properties.useBetaEndpoint` for now.

Comment (svenbru) [reply search-parts/src/dataSources/MicrosoftSearchDataSource.ts]: Makes sense, shouldn't the trimDuplicates property also be set only when useBetaEndpoint is enabled as this property is only exposed in the beta endpoint? I've included this in my latest commit.

Comment (wobba) [reply search-parts/src/dataSources/MicrosoftSearchDataSource.ts]: It should, and that one will hopefully/maybe never make it's way to v1.0. Thank you 🙂

**Comment by wobba** [comment]
There are some merge conflicts on loc files to resolve.

**Comment by svenbru** [comment]
Probably had to do with some auto formatting in VS Code, The conflicts should be corrected now.

**Pull request merged**